### ISSUE TITLE
subscription well_known_Tx check.

### ISSUE THREAD

**Issue by burdettadam** [Picolab/wrangler#119]
subscription should check inputs better, especially well known Tx. 
cid:161fc740e33cb971f162
![uploads/b3e01c5a-58fb-49ae-b798-63c070f101e4/image.png](https://s3-us-west-2.amazonaws.com/prod.huboard.com/uploads%2Fb3e01c5a-58fb-49ae-b798-63c070f101e4%2Fimage.png)

**Comment by LewisThomas** [comment]
I do not see a reason to strongly type check wellKnown's added to wrangler:subscription events, but perhaps raising a warning at incorrect formatting might be helpful for debugging.

**Comment by LewisThomas** [comment]
I opted to implement this as a strong check, with the assumption that channels will always maintain a length of 22 characters. Otherwise a formatting error event is raised. If this is too rigid a solution let me know.

Channel ID regex:
re#^[a-zA-Z0-9]{22}$#

https://github.com/Picolab/pico-engine/commit/e9421e52a2ed284b1d14f9fb65e92b7e200aa794

**Comment by windley** [comment]
That’s a good question. Check with Matthew on the DID generation code and see what he thinks it will do? 

Is there a reason to be so specific?

**Comment by LewisThomas** [comment]
Yes, after thinking about this a strong check is way overboard. Also after checking the code it currently generates a string that is (usually) 21 or 22 characters. This outcome appears entirely dependent on the seed given to the encoding function (ran code can be seen here: https://github.com/Picolab/node-sovrin-did/blob/master/index.js).

Anyway, I reverted the change. There doesn't seem to be any reason why ECIs can't be of arbitrary length, or why the length would not be able to change in the future.
